sources/Algo/src/Algo.cpp

```diff
@@ -148,6 +157,29 @@ GeneratorDefinitionAlgorithm::isDiagramValid(const inputs::Generator& generator)
   return valid;
 }
 
+bool
+GeneratorDefinitionAlgorithm::IsOtherGeneratorConnectedBySwitches(const NodePtr& node) const {
+  auto vl = node->voltageLevel.lock();
```

Comment:
Are we sure this lock will give us back a valid shared pointer ?

Reply:
At this point, yes because life cycle of the nodes is the same as the voltage levels (Network manager)